### PR TITLE
 fix: allow empty but required resolution property for [Bugzilla] (#4102)

### PR DESCRIPTION
**Pull request by seetd**
This was tested against a test case with the bug id 1536094 as shown in the attached.
Screen Shot 2019-10-02 at 1 48 40 AM

The test was not committed since the bug can be modified in future with the resolution field populated which will break the test. There is a way to query the the Bugzilla API and search by status providing a limit of 1 in order to create a robust test. But that involves changing the behavior of the service that is not much to do just for the test. So I am leaving out the unit test for this change

### CODE REVIEW

File: services/bugzilla/bugzilla.service.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ const schema = Joi.object({
     .items(
       Joi.object({
         status: Joi.string().required(),
-        resolution: Joi.string().required(),
+        resolution: Joi.string().allow(''),
```
Comment (calebcartwright) [inline services/bugzilla/bugzilla.service.js]: I think we still want to have the `.required()` to ensure the field is there. 

Would you mind validing that something like the below still solves the bug?
```suggestion
        resolution: Joi.string().allow('').required(),
```

Comment (seetd) [reply services/bugzilla/bugzilla.service.js]: It worked and I updated the PR to `resolution: Joi.string().required().allow('')` to make it look closer to the status property. It works as `resolution: Joi.string().allow('').required()` as well if you want that

Comment (calebcartwright) [reply services/bugzilla/bugzilla.service.js]: Awesome, thanks for confirming. IIRC we typically put `.required()` on the end, though as you noted it doesn't make a function difference. 

It's up to you if you'd like to swap those 👍

Comment (seetd) [reply services/bugzilla/bugzilla.service.js]: I swapped them and I also squash the commits into a single one. Hope that does not mess up anything.

Comment (calebcartwright) [reply services/bugzilla/bugzilla.service.js]: > I swapped them and I also squash the commits into a single one. Hope that does not mess up anything.

Thanks, and no worries. We always squash on merge so feel free to structure your commits in whichever way works best for you/makes the changes easier to review

Comment (seetd) [reply services/bugzilla/bugzilla.service.js]: Thanks I am done for this PR now

**Comment by calebcartwright** [comment]
Fix is confirmed in the review app!

**Comment by calebcartwright** [comment]
Also, if you're ever in Discord, drop us a note and we'll add you to the Contributor role

**Pull request merged**